docs/book/src/topics/flannel.md

```diff
@@ -0,0 +1,61 @@
+# Flannel
+
+This document describes how to use [Flannel](https://github.com/flannel-io/flannel) as your CNI.
+
+## Modify the Cluster resources
+
+Before deploying the cluster, change the `KubeadmControlPlane` value at `spec.kubeadmConfigSpec.clusterConfiguration.controllerManager.extraArgs.allocate-node-cidrs` to `"true"`
+
+```yaml
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+spec:
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      controllerManager:
+        extraArgs:
+          allocate-node-cidrs: "true"
+```          
+
+### Modify Flannel config
+
+_NOTE_: This is based off of the instructions at: https://github.com/flannel-io/flannel#deploying-flannel-manually
+
+You need to make an adjustment to the default flannel configuration so that the CIDR inside your CAPZ cluster matches the Flannel Network CIDR.
+
+View your capi-cluster.yaml and make note of the Cluster Network CIDR Block.  For example: 
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/16
+```
+
+
+Download the file at `https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml` and modify the `kube-flannel-cfg` ConfigMap.
+Set the value at `data.net-conf.json.Network` value to match your Cluster Network CIDR Block.
+
+
+```bash
+wget https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+```
+
+Edit kube-flannel.yml and change this section so that the Network section matches your Cluster CIDR
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+data:
+  net-conf.json: |
+    {
+      "Network": "192.168.0.0/16",
+      "Backend": {
+        "Type": "vxlan"
+      }
+    }
+```
```

Comment:
I understand it's kind of implied to apply `kube-flannel.yml` after this step, should we make it explicit here for completeness?

Reply:
Works for me.  Will add.

Reply:
Done.